qt/widgets/common/src/MuonFitPropertyBrowser.cpp

```diff
@@ -348,6 +348,10 @@ void MuonFitPropertyBrowser::setWorkspaceName(const QString &wsName) {
 * @param prop :: A pointer to the function name property
 */
 void MuonFitPropertyBrowser::enumChanged(QtProperty *prop) {
+  if (m_workspaceNames.empty() && this->isVisible()) {
+    g_log.error("No Data available. Please load Some data.");
+    return;
+  }
```

Comment:
This is still not quite right. I think we always want to return when `m_workspaceNames` is empty but we only want to print the error if the `this` is visible AND `m_workspaceNames` is empty. 

i.e.
```cpp
if(m_workspaceNames.empty()) {
    if (this->visible()))
        g_log.error("...");
    return;
}
```